test/Model/Reference/UnknownReferenceTest.php

```diff
@@ -0,0 +1,110 @@
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertInstanceOf(Reference::class, $reference);
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_an_id()
+    {
+        $reference = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertSame('id', $reference->getId());
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_a_date()
+    {
+        $reference = new UnknownReference('id', $date = new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertEquals($date, $reference->getDate());
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_authors()
+    {
+        $reference = new UnknownReference('id', new ReferenceDate(2000),
+            $authors = [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertEquals($authors, $reference->getAuthors());
+    }
+
+    /**
+     * @test
+     */
+    public function it_may_have_further_authors()
+    {
+        $with = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], true, 'title');
+        $withOut = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertTrue($with->authorsEtAl());
+        $this->assertFalse($withOut->authorsEtAl());
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_a_title()
+    {
+        $reference = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertSame('title', $reference->getTitle());
+    }
+
+    /**
+     * @test
+     */
+    public function it_may_have_details()
+    {
+        $with = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title', 'details');
+        $withOut = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title');
+
+        $this->assertSame('details', $with->getDetails());
+        $this->assertNull($withOut->getDetails());
+    }
+
+    /**
+     * @test
+     */
+    public function it_may_have_a_uri()
+    {
+        $with = new UnknownReference('id', new ReferenceDate(2000),
+            [new PersonAuthor(new PersonDetails('preferred name', 'index name'))], false, 'title', null, 'http://www.example.com/');
+        $withOut = new UnknownReference('id', new ReferenceDate(2000),
```

Comment:
any change of using a `Builder` instance for constructions with many irrelevant fields? This would be
```
$with = Builder::for(UnknownReference::class)->withUri('http://www.example.com/')->__invoke();
$withOut = Builder::for(UnknownReference::class)->withUri(null)->__invoke();
```